typescript-service-generator-core/src/main/java/com/palantir/code/ts/generator/ServiceClassParser.java

```diff
@@ -74,74 +76,89 @@ public ServiceModel parseServiceClass(Class<?> serviceClass, TypescriptServiceGe
         ModelCompiler compiler = new TypeScriptGenerator(settings.getSettings()).getModelCompiler();
 
         List<ServiceEndpointModel> endpointModels = Lists.newArrayList();
-        for (Method method : serviceMethods) {
-            endpointModels.add(computeEndpointModel(method, compiler, settings));
-        }
+        endpointModels = computeEndpointModels(serviceMethods, compiler, settings);
+
         Collections.sort(endpointModels);
         serviceModel.endpointModels(endpointModels);
         return serviceModel.build();
     }
 
-    private static ServiceEndpointModel computeEndpointModel(Method endpoint, ModelCompiler compiler, TypescriptServiceGeneratorConfiguration settings) {
-        ImmutableServiceEndpointModel.Builder ret = ImmutableServiceEndpointModel.builder();
-        ret.endpointName(endpoint.getName());
-        ret.javaReturnType(endpoint.getGenericReturnType());
-        ret.tsReturnType(compiler.typeFromJavaWithReplacement(endpoint.getGenericReturnType()));
-        ret.endpointMethodType(getMethodType(endpoint));
-
-        String annotationValue = "";
-        if (endpoint.getAnnotation(Path.class) != null) {
-            annotationValue = endpoint.getAnnotation(Path.class).value();
+    private static List<ServiceEndpointModel> computeEndpointModels(Set<Method> endpoints, ModelCompiler compiler, TypescriptServiceGeneratorConfiguration settings) {
+        Multimap<String, Method> endpointNameMap = ArrayListMultimap.create();
+        Map<Method, String> endpointNameGetter = Maps.newHashMap();
+        for (Method endpoint : endpoints) {
+            endpointNameMap.put(endpoint.getName(), endpoint);
+            endpointNameGetter.put(endpoint, endpoint.getName());
         }
-        ret.endpointPath(PathUtils.trimSlashes(annotationValue));
-        Consumes consumes = endpoint.getAnnotation(Consumes.class);
-        if (consumes != null) {
-            if (consumes.value().length > 1) {
-                throw new IllegalArgumentException("Don't know how to handle an endpoint with multiple consume types");
+        for (String endpointName : endpointNameMap.keySet()) {
+            List<Method> maybeDuplicates = Lists.newArrayList(endpointNameMap.get(endpointName).iterator());
+            if (maybeDuplicates != null && maybeDuplicates.size() > 1) {
+                endpointNameGetter.putAll(settings.duplicateEndpointNameResolver().resolveDuplicateNames(maybeDuplicates));
```

Comment:
does this mean if a client uses the default name resolution or doesn't correctly handle the methods, it will throw an NPE? maybe makes sense to provide a more specific error?